src/main/java/org/fundacionjala/virtualassistant/player/spotify/client/SpotifyClient.java

```diff
@@ -0,0 +1,242 @@
+            return root.path(ACCESS_TOKEN).asText();
+        } catch (JsonProcessingException e) {
+            throw new TokenExtractionException(ACCESS_TOKEN_JSON, e);
+        }
+    }
+
+    @Override
+    public String getSavedAlbums() {
+        return exchangeGetRequest(ApiMusic.SAVED_ALBUMS.url());
+    }
+
+    @Override
+    public String getSavedTracks() {
+        return exchangeGetRequest(ApiMusic.SAVED_TRACKS.url());
+    }
+
+    @Override
+    public String getFollowed() {
+        return exchangeGetRequest(ApiMusic.FOLLOWED.url());
+    }
+
+    @Override
+    public String getPlayerInfo() {
+        return exchangeGetRequest(ApiMusic.PLAYER_INFO.url());
+    }
+
+    private String exchangeGetRequest(String url){
+        return request.exchange(
+                getAccessToken(),
+                url,
+                HttpMethod.GET).getBody();
+    }
+
+    private HttpHeaders getHeader(){
+        return request.createHeader(getAccessToken());
+    }
+
+    @Override
+    public boolean playCurrentSong() {
+        HttpHeaders headers = getHeader();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        HttpStatus response = request.exchange(
+                headers,
+                ApiMusic.PLAY.url(),
+                HttpMethod.PUT
+        ).getStatusCode();
+        return response.equals(HttpStatus.OK);
+    }
+    public boolean playNextTrackOnDevice() {
+        HttpStatus response = request.exchange(
+                getAccessToken(),
+                ApiMusic.PLAY_NEXT_TRACK.url(),
+                HttpMethod.POST
+        ).getStatusCode();
+
+        return response.equals(HttpStatus.NO_CONTENT);
+    }
+
+    public boolean playPreviousTrackOnDevice() {
+        HttpStatus response = request.exchange(
+                getAccessToken(),
+                ApiMusic.PLAY_PREVIOUS_TRACK.url(),
+                HttpMethod.POST
+        ).getStatusCode();
+
+        return response.equals(HttpStatus.NO_CONTENT);
+    }
+
+    public boolean pauseSongOnDevice(String trackUri) {
+        HttpHeaders headers = getHeader();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+        String requestBody = CustomQuery.uriRequestBody(trackUri);
+        HttpEntity<String> entity = new HttpEntity<>(requestBody, headers);
+
+        HttpStatus response = request.exchange(
+                entity,
+                ApiMusic.PAUSE.url(),
+                HttpMethod.PUT
+        ).getStatusCode();
+        return response.equals(HttpStatus.OK);
+    }
+
+    public String searchTrackByArtistAndTrack(String artist, String track) {
+        String searchQuery = CustomQuery.searchRequestBody(artist, track);
+        ResponseEntity<String> response = request.exchange(
+                getAccessToken(),
+                searchQuery,
+                HttpMethod.GET
+        );
+
+        if (response.getStatusCode().equals(HttpStatus.OK)) {
```

Comment:
here please invert the comparison, so if you get a null value on the response it wont thrown a null pointer exception
`HttpStatus.OK.equals(response.getStatusCode())`